core/utils.go

```diff
@@ -73,3 +78,44 @@ func makeMessageSignatureVerifier(authen api.Authenticator) messageSignatureVeri
 func isPrimary(view uint64, id uint32, n uint32) bool {
 	return uint64(id) == view%uint64(n)
 }
+
+func messageString(msg interface{}) string {
+	switch msg := msg.(type) {
+	case *messages.Request:
+		m := msg.GetMsg()
+		return fmt.Sprintf("REQUEST<client=%d seq=%d payload=%s>",
+			m.GetClientId(), m.GetSeq(), m.GetPayload())
+	case *messages.Reply:
+		m := msg.GetMsg()
+		return fmt.Sprintf("REPLY<replica=%d seq=%d result=%s>",
+			m.GetReplicaId(), m.GetSeq(), m.GetResult())
+	case *messages.Prepare:
+		m := msg.GetMsg()
+		req := m.GetRequest().GetMsg()
+		return fmt.Sprintf("PREPARE<replica=%d view=%d client=%d seq=%d>",
+			m.GetReplicaId(), m.GetView(),
+			req.GetClientId(), req.GetSeq())
+	case *messages.Commit:
+		m := msg.GetMsg()
+		req := m.GetRequest().GetMsg()
+		return fmt.Sprintf("COMMIT<replica=%d primary=%d view=%d client=%d seq=%d>",
+			m.GetReplicaId(), m.GetPrimaryId(), m.GetView(),
+			req.GetClientId(), req.GetSeq())
+	}
+	return "(unknown message)"
+}
```

Comment:
Does it make sense to include UI value in log output? I think that it's valuable info for developers/debuggers.